### PR TITLE
fix: blacklist unknown xgboost image versions

### PR DESCRIPTION
**Pull request by edwardjkim**
*Issue #, if available:*

*Description of changes:*

When an unknown version is passed to the `repo_version` parameter of [get_image_uri](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/amazon/amazon_estimator.py#L610), the function prints a misleading warning and returns a completely wrong image URI.

```
print(get_image_uri('us-west-2', 'xgboost', '1.1-1'))
```
```
WARNING:root:There is a more up to date SageMaker XGBoost image. To use the newer image, please set 'repo_version'='1.0-1'. For example:
	get_image_uri(region, 'xgboost', '1.0-1').
```
```
'433757028032.dkr.ecr.us-west-2.amazonaws.com/xgboost:1.1-1'
```

Not only `1.1-1` is currently unavailable, the account `433757028032` and repository `xgboost` are for 1p xgboost, not OS XGBoost. This is an un undesired behavior.

Desired behavior:

- Fix above behavior to return invalid version number with appropriate error message
- If version is valid but not latest then above warning should be printed.
- The container returned should be 1P XGBoost only when no version number is provided.

This PR changes `get_image_uri()` to blacklist unknown xgboost image versions and adds appropriate unit tests.

*Testing done:*

`export IGNORE_COVERAGE=- ; tox -e black-format,flake8,pylint,py36 tests/unit/test_amazon_estimator.py; unset IGNORE_COVERAGE`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to any/all clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by edwardjkim** [comment]
@balajitummala @aws-patlin Can you please review?

File: tests/unit/test_amazon_estimator.py [hunk 1]
```diff
@@ -484,7 +529,7 @@ def test_is_latest_xgboost_version():
             assert _is_latest_xgboost_version(version) is False
 
     assert _is_latest_xgboost_version("0.90-1-cpu-py3") is False
-
+    assert _is_latest_xgboost_version("0.90-2-cpu-py3") is False
```
Comment (aws-patlin) [inline tests/unit/test_amazon_estimator.py:532]: Can we just perform a loop over the supported versions, excluding the last entry? That way we don't need to add a new line each time there's a new version.

File: src/sagemaker/amazon/amazon_estimator.py [hunk 2]
```diff
@@ -621,41 +621,67 @@ def get_image_uri(region_name, repo_name, repo_version=1):
...
+        if "-" not in repo_version:
+            xgboost_version_matches = [
+                version
+                for version in XGBOOST_SUPPORTED_VERSIONS
+                if repo_version == version.split("-")[0]
+            ]
+            if xgboost_version_matches:
+                # Assumes that XGBOOST_SUPPORTED_VERSION is sorted from oldest version to latest,
+                # and the latest version is at the end of the list.
+                repo_version = xgboost_version_matches[-1]
```
[the remark this answers is not in the document]
Comment (edwardjkim) [reply src/sagemaker/amazon/amazon_estimator.py:641]: I think you are right. We don't want to change 0.90 -> 0.90-2 if customers were getting 0.90-1 previously. Will update.

**Comment by aws-patlin** [comment]
Looks like there's a formatting issue in one of the files. You'll need to run black to fix it.

**Comment by edwardjkim** [comment]
Thanks! @aws-patlin

**Comment by laurenyu** [comment]
looks like @aws-patlin approved it earlier. let me know if you want any other approvals; otherwise, I'll treat it as ready to be merged in.

**Pull request merged**